run_project_tests.py

```diff
@@ -528,12 +538,10 @@ def have_objcpp_compiler():
             return False
     return True
 
-def have_java():
-    if shutil.which('javac') and shutil.which('java'):
-        return True
-    return False
+def have_java() -> bool:
+    return bool(shutil.which('javac') and shutil.which('java'))
```

Comment:
no need to wrap this in `bool`, `and` is a logical operator, it always returns bool

Reply:
```python
'a' and 'b' == 'b'
```
a holiday syntax surprise from Python

Reply:
...discovered thanks to refactoring and type annotation

Reply:
`and` and `or` return the first value that unambiguously answers the logical, not a boolean.

Reply:
```python
def have_java() -> str:
    return shutil.which('javac') and shutil.which('java')
```